File: src/main/java/engine/feature/shader/ShaderCreator.java
package engine.feature.shader;

import com.jogamp.opengl.GL4;
import org.jetbrains.annotations.NotNull;

public interface ShaderCreator {

    @NotNull
    Shader create(@NotNull String vertexSource, @NotNull String fragmentSource, @NotNull GL4 gl);

    @NotNull
    Shader create(@NotNull String vertexSource, @NotNull String fragmentSource, @NotNull String geometrySource, @NotNull GL4 gl);
}

File: src/main/java/engine/feature/shader/ShaderCreator.kt
package engine.feature.shader

import com.jogamp.opengl.GL4

interface ShaderCreator {
    fun create(vertexSource: String, fragmentSource: String, gl: GL4): Shader
    fun create(vertexSource: String, fragmentSource: String, geometrySource: String, gl: GL4): Shader
}
